shadowsocks/udp.go

```diff
@@ -151,12 +146,17 @@ func (s *udpService) Start() {
 
 			payload := buf[len(tgtAddr):]
 
-			targetConn := nm.Get(clientAddr.String())
+			targetConn, clientLocation := nm.Get(clientAddr.String())
 			if targetConn == nil {
 				targetConn, err = net.ListenPacket("udp", "")
 				if err != nil {
 					return onet.NewConnectionError("ERR_CREATE_SOCKET", "Failed to create UDP socket", err)
```

Comment:
It seems this error is being reported without location. It may explain why I'm seeing UDP traffic without location in the metrics. We should move the GetLocation call to before the ListenPacket call.